### ISSUE TITLE
refactor(http): only use readyState in mock connection

### ISSUE THREAD

**Issue by jeffbcross** [angular/angular#2799]
`Connection` instances have a readyState property, with corresponding `ReadyState` enum with values borrowed from `XMLHttpRequest` [readyState](http://www.w3.org/TR/XMLHttpRequest/#dom-xmlhttprequest-readystate) spec. The readyState is useful for tests to easily determine when connections have been fulfilled or not. All use cases for watching `readyState` changes to provide a good user experience should be able to be satisfied with observers provided to `Http`, as well as the Observables/Streams returned from `Http`.

This property should be limited to `MockConnection`, renamed to `state`, and changed to reflect states that Http cares about, namely "Sent", "Downloading", and "Completed".

**Comment by live2trance** [comment]
I've started to tackle this one but i've a question. 
I've added the new states to http enums, removed the readyState property from the Connection interface and renamed it under the mock_backend to state. But now under the Jsonp_backend i've some logic that checks for the cancelled state under the state property, but i don't have it under the states and i don't know exactly how could i check it. How can i properly check if the connection has been cancelled?
